lib/Common/Core/SysInfo.cpp

```diff
@@ -8,6 +8,7 @@
 #endif
 #include <wincrypt.h>
 #include <VersionHelpers.h>
+#include <type_traits>
```

Comment:
>   Instead of dealing with counter, can we use the same static initialization approach (which is also thread safe) ?

You mean using construction on first use? If so, the only reason I didn't do that was because it's used everywhere in the codebase, and I was trying to keep the diff small.

> Is there a reason why don't we host this under PAL ?

You tell me, I'm happy to move it if you think it's better.

> Why we do limit this only for static library?

 For dynamic libraries, it fails to link, since `ch` includes the header, but doesn't link against the implementation.